From a4ece788666ac49f84adf0a3b91e3d733fd71880 Mon Sep 17 00:00:00 2001
From: Balamurugan Selvarajan <quic_bselvara@quicinc.com>
Date: Tue, 3 May 2022 13:19:14 +0530
Subject: [PATCH] ath12k: Introduced module param to En/Dis FW segmented memory

Introduced module parameter to enable/disable FW segmented memory.
To enable FW segmented memory, fw_mem_seg  should be set to
1 during insmod. "ex insmod ath12k.ko fw_mem_seg=1"

Signed-off-by: Balamurugan Selvarajan <quic_bselvara@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/pci.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/pci.c b/drivers/net/wireless/ath/ath12k/pci.c
index 225ad2e..4dba3da 100644
--- a/drivers/net/wireless/ath/ath12k/pci.c
+++ b/drivers/net/wireless/ath/ath12k/pci.c
@@ -42,6 +42,11 @@
 #define PCIE_PCIE_LOCAL_REG_PCIE_LOCAL_RSV0	0x1E03164
 #define QRTR_INSTANCE_MASK			0x000000FF
 
+unsigned int ath12k_fw_mem_seg;
+EXPORT_SYMBOL(ath12k_fw_mem_seg);
+module_param_named(fw_mem_seg, ath12k_fw_mem_seg, uint, 0644);
+MODULE_PARM_DESC(fw_mem_seg, "Enable/Disable FW segmentted memory");
+
 static const struct pci_device_id ath12k_pci_id_table[] = {
 	{ PCI_VDEVICE(QCOM, QCN9274_DEVICE_ID) },
 	{0}
@@ -1306,6 +1311,10 @@ static int ath12k_pci_probe(struct pci_dev *pdev,
 	if (!of_property_read_u32(ab->dev->of_node, "memory-region", &addr))
 		ab->bus_params.fixed_mem_region = true;
 
+	/* This is HACK to bring up the qcn9224 with segemnted memory */
+	if (ath12k_fw_mem_seg)
+		ab->bus_params.fixed_mem_region = false;
+
 	ret = ath12k_pci_claim(ab_pci, pdev);
 	if (ret) {
 		ath12k_err(ab, "failed to claim device: %d\n", ret);
-- 
2.17.1

